tmt/steps/provision/beaker_mrack.py

```diff
@@ -0,0 +1,532 @@
+
+        if self.guestname is None:
+            return
+
+        self.api.delete()
+
+
+@tmt.steps.provides_method('beaker')
+class ProvisionBeaker(tmt.steps.provision.ProvisionPlugin):
+    """
+    Provision guest on Beaker system using mrack
+
+    Minimal configuration could look like this:
+
+        provision:
+            how: beaker
+            image: Fedora
+
+    Full configuration example: TBA
+
+    """
+
+    _data_class = ProvisionBeakerData
+    _guest_class = GuestBeaker
+
+    # Guest instance
+    _guest = None
+
+    _keys = [
+        'arch',
+        'image',
+        'hardware',
+        'pool',
+        'priority-group',
+        'keyname',
+        'user-data',
+        'provision-timeout',
+        'provision-tick',
+        ]
+
+    @classmethod
+    def options(cls, how: Optional[str] = None) -> List[tmt.options.ClickOptionDecoratorType]:
+        """ Prepare command line options for Artemis """
+        return cast(List[tmt.options.ClickOptionDecoratorType], [
+            click.option(
+                '--arch', metavar='ARCH',
+                help='Architecture to provision.'
+                ),
+            click.option(
+                '--image', metavar='COMPOSE',
+                help='Image (distro or "compose" in Beaker terminology) '
+                     'to provision.'
+                ),
+            click.option(
+                '--keyname', metavar='NAME',
+                help='SSH key name.'
+                ),
+            click.option(
+                '--provision-timeout', metavar='SECONDS',
+                help=f'How long to wait for provisioning to complete, '
+                     f'{DEFAULT_PROVISION_TIMEOUT} seconds by default.'
+                ),
+            click.option(
+                '--provision-tick', metavar='SECONDS',
+                help=f'How often check Beaker for provisioning status, '
+                     f'{DEFAULT_PROVISION_TICK} seconds by default.',
+                ),
+            ]) + super().options(how)
+
+    def default(self, option: str, default: Optional[Any] = None) -> Any:
+        """ Return default data for given option """
+
+        return getattr(BeakerGuestData(), option.replace('-', '_'), default)
+
+    def wake(self, data: Optional[BeakerGuestData] = None) -> None:  # type: ignore[override]
+        """ Wake up the plugin, process data, apply options """
+
+        super().wake(data=data)
+
+        if data:
+            self._guest = GuestBeaker(data=data, name=self.name, parent=self.step)
+
+    def go(self) -> None:
+        """ Provision the guest """
+        super().go()
+
+        try:
+            user_data = {
+                key.strip(): value.strip()
+                for key, value in (
+                    pair.split('=', 1)
+                    for pair in self.get('user-data')
+                    )
+                }
+
+        except ValueError:
+            raise ProvisionError('Cannot parse user-data.')
```

Comment:
User data is Artemis' thing, not needed for Beaker.

Reply:
Hmm, or does the plugin use it? I see there's `mrack_req.update({"user_data": environment.get("user_data")})` - Artemis sends this key/value mapping to Artemis service, where it's attached to a guest request.

Reply:
removed :)